pkg/installer/check.go

```diff
@@ -227,8 +227,8 @@ func (o *Installer) ProviderURLParse(path string, acceptPreRelease bool) (packag
 
 					var buf bytes.Buffer
 					if err = tmp.Execute(&buf, hdPkg); err == nil {
-						packageURL = fmt.Sprintf("https://github.com/%s/%s/releases/download/%s/%s",
-							o.Org, o.Repo, version, buf.String())
+						packageURL = fmt.Sprintf("https://github.com/%s/%s/releases/download/%s/%s.%s",
+							o.Org, o.Repo, version, buf.String(), getPackagingFormat(o))
```

Comment:
`buf` is a template value, it allows users to have a more flexible filename. See the following errors:

```
Error: cannot download from https://github.com/kubesphere-sigs/ks/releases/download/v0.0.52/ks-linux-amd64.tar.gz., response error: failed to download from 'https://github.com/kubesphere-sigs/ks/releases/download/v0.0.52/ks-linux-amd64.tar.gz.': status code: 404, content length error: <nil>
```

In [this case](https://github.com/LinuxSuRen/hd-home/blob/master/config/kubesphere-sigs/ks.yml), there is no `package` defined.

Reply:
I have made a fix for this. Does the fix seem reasonable? WDYT @LinuxSuRen?